src/analytics/AnalyticsManager.ts

```diff
@@ -0,0 +1,497 @@
+    }
+
+    // Track session start
+    this.trackSessionStart();
+  }
+
+  /**
+   * Register global properties that are sent with every event
+   */
+  private registerGlobalProperties(): void {
+    const globalProps: Record<string, unknown> = {
+      sdkVersion: this.sdkVersion,
+      sessionId: this.session.getSessionId(),
+    };
+
+    if (this.network) {
+      globalProps.network = this.network;
+    }
+
+    if (this.config.appMetadata?.appName) {
+      globalProps.appName = this.config.appMetadata.appName;
+    }
+
+    if (this.config.appMetadata?.appVersion) {
+      globalProps.appVersion = this.config.appMetadata.appVersion;
+    }
+
+    if (this.config.appMetadata?.appUrl) {
+      globalProps.appUrl = this.config.appMetadata.appUrl;
+    }
+
+    // Add environment if in browser
+    if (typeof window !== 'undefined') {
+      globalProps.environment = this.detectEnvironment();
+    }
+
+    this.provider.register(globalProps);
+
+    if (this.config.debug) {
+      console.log('[AnalyticsManager] Global properties registered:', globalProps);
+    }
+  }
+
+  /**
+   * Detect environment (development/production)
+   */
+  private detectEnvironment(): string {
+    if (typeof window === 'undefined') {
+      return 'server';
+    }
+
+    // Check common environment indicators
+    if (window.location.hostname === 'localhost' || window.location.hostname === '127.0.0.1') {
+      return 'development';
+    }
+
+    return 'production';
+  }
+
+  /**
+   * Track session start
+   */
+  private trackSessionStart(): void {
+    if (!this.enabled) return;
+
+    const url = typeof window !== 'undefined' ? window.location.href : undefined;
+    const referrer = typeof document !== 'undefined' ? document.referrer || undefined : undefined;
+
+    this.track(
+      NexusAnalyticsEvents.SESSION_STARTED,
+      this.session.getSessionStartProperties(url, referrer) as Record<string, unknown>
+    );
+
+    if (this.config.debug) {
+      console.log('[AnalyticsManager] Session started:', this.session.getSessionId());
+    }
+  }
+
+  /**
+   * Sanitize properties by removing sensitive financial data if privacy.anonymizeAmounts is enabled
+   */
+  private sanitizeProperties(properties?: Record<string, unknown>): Record<string, unknown> {
+    if (!properties || !this.config.privacy?.anonymizeAmounts) {
+      return properties || {};
+    }
+
+    // Fields to remove when anonymizing amounts
+    const sensitiveFields = ['amount', 'valueUsd', 'totalValueUsd', 'fees', 'gasUsed', 'value'];
+
+    const sanitized = Object.create(null);
+    for (const [key, value] of Object.entries(properties)) {
+      if (!sensitiveFields.includes(key)) {
+        sanitized[key] = value;
+      }
+    }
+
+    return sanitized;
+  }
+
+  /**
+   * Validate properties to prevent prototype pollution and injection attacks
+   */
+  private validateProperties(properties: Record<string, unknown>): Record<string, unknown> {
```

Comment:
Does this function needs to be a method of `AnalyticsManager`?
Or do idea is to bundle all `AnalyticsManager` stuff into one class?